mark-kavdi/server.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+
+// DONE: Install and require the NPM Postgres package 'pg' into your server.js, and ensure that it is then listed as a dependency in your package.json
+const fs = require('fs');
+const express = require('express');
+const PG = require('pg');
+
+// REVIEW: Require in body-parser for post requests in our server. If you want to know more about what this does, read the docs!
+const bodyParser = require('body-parser');
+const PORT = process.env.PORT || 3000;
+const app = express();
+
+// DONE: Complete the connection string for the url that will connect to your local postgres database
+// Windows and Linux users; You should have retained the user/pw from the pre-work for this course.
+// Your url may require that it's composed of additional information including user and password
+// const conString = 'postgres://USER:PASSWORD@HOST:PORT/DBNAME';
+const conString = 'postgres://localhost:5432/articles';
+
+// DONE: Our pg module has a Client constructor that accepts one argument: the conString we just defined.
+//       This is how it knows the URL and, for Windows and Linux users, our username and password for our
+//       database when client.connect is called on line 26. Thus, we need to pass our conString into our
+//       pg.Client() call.
+const client = new pg.Client(conString);
+
+// REVIEW: Use the client object to connect to our DB.
+client.connect();
+
+
+// REVIEW: Install the middleware plugins so that our app is aware and can use the body-parser module
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({extended: true}));
+app.use(express.static('./public'));
+
+
+// REVIEW: Routes for requesting HTML resources
+app.get('/new', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // DONE: Numbers 4 and 5 correspond to the following line of code.
+  response.sendFile('new.html', {root: './public'});
+});
+
+
+// REVIEW: Routes for making API calls to use CRUD Operations on our database
+app.get('/articles', function(request, response) {
+  // COMMENT: What number(s) of the full-stack-diagram.png image correspond to the following line of code? Which method of article.js is interacting with this particular piece of `server.js`? What part of CRUD is being enacted/managed by this particular piece of code?
+  // DONE: 2, 3, 4, and 5 are used in this get request which is interacting with the fetchAll method in article.js as part of the Read cycle of CRUD.
```

Comment:
Technically, this does not correspond to 2, since this code is not making the request itself.